src/main/java/com/comcast/zucchini/AbstractZucchiniTest.java

```diff
@@ -31,18 +37,34 @@
  */
 public abstract class AbstractZucchiniTest {
 
-    private static Logger logger = LoggerFactory.getLogger(AbstractZucchiniTest.class);
-    private TestNGZucchiniRunner runner;
-    static HashMap<String, JsonArray> featureSet = new HashMap<String, JsonArray>();
+    private static final Logger LOGGER = LoggerFactory.getLogger(AbstractZucchiniTest.class);
+    static Map<String, JsonArray> featureSet = new HashMap<String, JsonArray>();
 
     /* Synchronization and global variables.  DO NOT TOUCH! */
+    private static Object lock = new Object();
     private static Boolean hooked = false;
 
+    /* store the list of contexts here */
+    List<TestContext> contexts;
```

Comment:
I'll take a look to check and be sure that I'm using it though.  I'm pretty sure that I use it in sizing the barriers auto-magically.